doc/operator_auth.md

```diff
@@ -0,0 +1,163 @@
+operation using the temporary key without requiring ongoing involvement from the user's wallet.
+
+The generated public key is converted into a JSON structure with additional metadata:
+* `pubkey`: The public key information.
+* `alg`: The signing algorithm, ECDSA.
+* `domain`: The domain for which the key is valid.
+* `address`: The Ethereum address of the sender, binding the public key to this identity.
+* `expires`: The expiration time of the key.
+
+Example
+```json
+{
+  "pubkey": {
+    "crv": "P-256",
+    "kty": "EC",
+    "x": "hbslLmhG3h2RwuzBYNVeQ7WCbU-tUzMjSpCFO2i5-tA",
+    "y": "KI4FJARKwyYcRy6xz1J9lu8OItV87Fw91eThe2hnnuc"
+  },
+  "alg": "ECDSA",
+  "domain": "localhost",
+  "address": "0x8Dd070629F107e7946dD68BDcb8ABE8475F47B0E",
+  "expires": "2010-12-26T17:05:55Z"
+}
+```
+
+This public key is signed using the Ethereum account to ensure its authenticity. The resulting signature is
+combined with the public key into a payload and sent as the `X-SignedPubKey` header.
+
+### 2. Sign Operation Payload
+
+#### Operation Payload Format
+
+The operation payload is a JSON object that encapsulates the details of an API request. It ensures that the request's
+integrity can be verified through signing. Below are the fields included:
+
+- **`time`**: (string, ISO 8601 format) The timestamp for when the operation is valid, including the timezone is mandatory (`Z`
+  indicates UTC). This helps prevent replay attacks (capturing the packet and replying it multiple time). e.g. `"2010-12-25T17:05:55Z"`
+- **`method`**: (string) The HTTP method used for the operation (e.g., `GET`, `POST`).
+- **`path`**: (string) The endpoint path of the request (e.g., `/`).
+- **`domain`**: (string) The domain associated with the request. This ensures the request is valid for the intended
+  CRN. (e.g., `localhost`).
+
+Example
+
+```json
+{
+  "time": "2010-12-25T17:05:55Z",
+  "method": "GET",
+  "path": "/",
+  "domain": "localhost"
+}
+```
+
+It is sent serialized as a hex string.
+
+#### Signature
+This payload is serialized, signed, and sent in the `X-SignedOperation` header to ensure the integrity and authenticity
+of the request.
+
+* The operation payload (containing details such as time, method, path, and domain) is serialized and converted into a byte array.
+* The JWK (private key) is used to sign this operation payload, ensuring its integrity. This signature is then included in the X-SignedOperation header.
+
+
+###  3. Include authentication Headers
+These two headers are to be added to the HTTP Request:
+
+1. **`X-SignedPubKey` Header:**
+    - This header contains the public key payload and the signature of the public key generated by the Ethereum account.
+
+   Example:
+   ```json
+   {
+      "payload": "<hexadecimal string of the public key payload>",
+      "signature": "<Ethereum signed public key>"
+   }
+   ```
+
+2. **`X-SignedOperation` Header:**
```

Comment:
Is it normal I get text highlighted in red on GitHub ?

![image](https://github.com/user-attachments/assets/f34c916f-d87e-4246-baf1-f22c580ca12b)

Reply:
saw it not sure why it happend, is red in the diff but in the final version it looks correct

![image](https://github.com/user-attachments/assets/2f1edb28-d51e-4044-8f8a-597c25bea9fb)